pysat/_meta.py

```diff
@@ -111,93 +72,89 @@ class Meta(object):
+                                      'long_name': [None, 'boohoo']}
 
         # assign from another Meta object
         meta[key1] = meta2[key2]
 
         # access fill info for a variable, presuming default label
         meta[key1, 'fill']
+
         # access same info, even if 'fill' not used to label fill values
         meta[key1, meta.fill_label]
 
 
         # change a label used by Meta object
         # note that all instances of fill_label
-        # within the meta object are updated
-        meta.fill_label = '_FillValue'
-        meta.plot_label = 'Special Plot Variable'
+        # within the meta object are NOT updated FIX??
+        meta.labels.fill_val = '_FillValue'
+        meta.labels.plot = 'Special Plot Variable'
+
         # this feature is useful when converting metadata within pysat
         # so that it is consistent with externally imposed file standards
 
     """
 
-    def __init__(self, metadata=None, units_label='units',
-                 name_label='long_name', notes_label='notes',
-                 desc_label='desc', plot_label='label', axis_label='axis',
-                 scale_label='scale', min_label='value_min',
-                 max_label='value_max', fill_label='fill',
-                 export_nan=None):
+    # -----------------------------------------------------------------------
+    # Define the magic methods
 
-        if export_nan is None:
-            export_nan = []
+    def __init__(self, metadata=None,
+                 labels={'units': ('units', str), 'name': ('long_name', str),
+                         'notes': ('notes', str), 'desc': ('desc', str),
+                         'plot': ('plot', str), 'axis': ('axis', str),
+                         'scale': ('scale', str),
+                         'min_val': ('value_min', float),
+                         'max_val': ('value_max', float),
+                         'fill_val': ('fill', float)}, export_nan=[]):
 
-        # set mutability of Meta attributes
+        # Set mutability of Meta attributes.  This flag must be set before
+        # anything else, or `__setattr__` breaks.
         self.mutable = True
 
-        # set units and name labels directly
-        self._units_label = units_label
-        self._name_label = name_label
-        self._notes_label = notes_label
-        self._desc_label = desc_label
-        self._plot_label = plot_label
-        self._axis_label = axis_label
-        self._scale_label = scale_label
-        self._min_label = min_label
-        self._max_label = max_label
-        self._fill_label = fill_label
-
-        # by default metadata with a value of nan will not be exported
-        # unless the name is in the _export_nan list. Initialize the list
-        # with the fill label, since it is reasonable to assume that a fill
-        # value of nan would be intended to be exported
-        self._export_nan = [fill_label] + export_nan
+        # Set the NaN export list
+        self._export_nan = export_nan
```

Comment:
Why did you remove fill_val as a default for NaN support?

Reply:
Originally I did not see what this did, so I had removed it and added it back in.  As is, any float value should all have NaN support.